process/block/metablockProposal.go

```diff
@@ -1089,20 +1093,14 @@ func (mp *metaProcessor) getPreviousExecutedBlock() data.HeaderHandler {
 	return blockHeader
 }
 
-func (mp *metaProcessor) computeEpochStartDataForHeader(metaHeader data.MetaHeaderHandler) (*block.EpochStart, error) {
+func (mp *metaProcessor) getComputedEpochStartData() (*block.EpochStart, error) {
 	mp.mutEpochStartData.RLock()
 	defer mp.mutEpochStartData.RUnlock()
 
 	if mp.epochStartDataWrapper == nil ||
 		mp.epochStartDataWrapper.EpochStartData == nil {
 		return nil, process.ErrNilEpochStartData
 	}
-	if metaHeader.IsStartOfEpochBlock() && mp.epochStartDataWrapper.Epoch != metaHeader.GetEpoch() {
-		return nil, process.ErrEpochDoesNotMatch
-	}
-	if len(mp.epochStartDataWrapper.EpochStartData.LastFinalizedHeaders) == 0 {
```

Comment:
so for epoch start block we don't check this anymore?

Reply:
it is now checked with checkEpochCorrectnessV3 which is more comprehensive
it has also a isEpochStartBlock := headerHandler.IsStartOfEpochBlock()
which verifies that len is > 0